modules/openshift-developer-cli-basic-odo-cli-commands.adoc

```diff
@@ -0,0 +1,522 @@
+
+----
+  # Log out
+  odo logout
+----
+
+
+== preference
+
+Modify `odo` specific configuration settings within the global preference file.
+
+.Example using preference
+----
+
+  # For viewing the current preferences
+  odo preference view
+  
+  # Set a preference value in the global preference
+  odo preference set UpdateNotification false
+  odo preference set NamePrefix "app"
+  odo preference set Timeout 20
+
+  # Unset a preference value in the global preference
+  odo preference unset  UpdateNotification
+  odo preference unset  NamePrefix
+  odo preference unset  Timeout
+----
+ 
+[NOTE]
+====
+By default, the path to the global preference file is `~/.odo/preferece.yaml` and it is stored in the environment variable `GLOBALODOCONFIG`. You can set up a custom path by setting the value of the environment variable to a new preference path, for example `GLOBALODOCONFIG="new_path/preference.yaml"`
+====
+
+.Available Parameters:
+|=======
+|NamePrefix |The default prefix is the current directory name. Use this value to set a default name prefix.
+|Timeout |The timeout (in seconds) for OpenShift server connection checks.
+|UpdateNotification |Controls whether an update notification is shown.
+|=======
+
+== project
+
+Perform project operations.
+
+.Example using project
+
+----
+  # Set the active project
+  odo project set
+
+  # Create a new project
+  odo project create myproject
+
+  # List all the projects
+  odo project list
+
+  # Delete a project
+  odo project delete myproject
+
+  # Get the active project
+  odo project get
+----
+
+== push
+
+Push source code to a component.
+
+.Example using push
+
+----
+  # Push source code to the current component
+  odo push
+  
+  # Push data to the current component from the original source.
+  odo push
+  
+  # Push source code in ~/mycode to component called my-component
+  odo push my-component --context ~/mycode
+----
+
+
+== service
+
+Perform service catalog operations.
+
+.Example using service
+
+----
+  # Create new postgresql service from service catalog using dev plan and name my-postgresql-db.
+  odo service create dh-postgresql-apb my-postgresql-db --plan dev -p postgresql_user=luke -p postgresql_password=secret
+
+  # Delete the service named 'mysql-persistent'
+  odo service delete mysql-persistent
+
+  # List all services in the application
+  odo service list
+----
+
+
+== storage
+
+Perform storage operations.
+
+.Example using storage
+
+----
+  # Create storage of size 1Gb to a component
+  odo storage create mystorage --path=/opt/app-root/src/storage/ --size=1Gi
+  # Delete storage mystorage from the currently active component
+  odo storage delete mystorage
+  
+  # Delete storage mystorage from component 'mongodb'
+  odo storage delete mystorage --component mongodb
+  # List all storage attached or mounted to the current component and
+  # all unattached or unmounted storage in the current application
+  odo storage list
+----
+
+== unlink
+
+.Example using unlink
```

Comment:
The sentence on line 422 (Unlink a component or a service.) should be added before the example, to keep it consistent with the rest of the doc